### PR TITLE
Pgen list

### PR DESCRIPTION
**Pull request by sarahmarshy**
Can now list all available tools and targets. Export creates a project file. Will need to revise handling of group names to add all files to project.

### CODE REVIEW

**Comment by 0xc0170** [comment]
Could they be split? One list extension and the other one yaml generation improvements?
Now it's fine as it is. I'll scan through changes

File: project_generator/project.py [hunk 1]
```diff
@@ -525,52 +534,66 @@ def copy_files(self, proj_dic, tool):
...
-        include_filetypes = FILES_EXTENSIONS['include_paths']
-        lib_filetypes = FILES_EXTENSIONS['source_files_lib']
-        obj_filetypes = FILES_EXTENSIONS['source_files_obj']
+    def determine_tool(linker_ext):
+        if "sct" in linker_ext or "lin" in linker_ext:
+            return "uvision"
+        elif "ld" in linker_ext:
+            return "make_gcc_arm"
+        elif "icf" in linker_ext:
+            return "IAR"
```
Comment (0xc0170) [inline project_generator/project.py]: should we use lower case for tools?
It  would be return "iar". For ld should return gcc. This means that this is not tool as pgen understand it - isn't  this toolchain? Becuase tool can be make_gcc_arm or eclipse_gcc_project or uvision_gcc.

Comment (sarahmarshy) [inline project_generator/project.py]: Yes, you're right. Changing. Will commit in a moment.

**Comment by 0xc0170** [comment]
LGTM

**Comment by 0xc0170** [comment]
A change here:

```
group_names = project_file_data['common']['sources'].keys()
```
This assumes there are groups (sources is a dic), which might not be true - I got a project where this is false, then receiving AttributeError - AttributeError: 'list' object has no attribute 'keys'. This error should be handled, and group_names then should be empty.

**Pull request merged**